mattermost-plugin/server/data_retention_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/mattermost/focalboard/server/server"
+	"github.com/mattermost/focalboard/server/services/config"
+	"github.com/mattermost/focalboard/server/services/permissions/localpermissions"
+	"github.com/mattermost/focalboard/server/services/store/mockstore"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/stretchr/testify/assert"
+)
+
+type TestHelperMockStore struct {
+	Server *server.Server
+	Store  *mockstore.MockStore
+}
+
+func SetupTestHelperMockStore(t *testing.T) *TestHelperMockStore {
+	th := &TestHelperMockStore{}
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockStore := mockstore.NewMockStore(ctrl)
+
+	th.Server = newTestServerMock(mockStore)
+	th.Store = mockStore
+
+	return th
+}
+
+func newTestServerMock(mockStore *mockstore.MockStore) *server.Server {
+	config := &config.Configuration{
+		EnableDataRetention: false,
+		DataRetentionDays:   10,
+		FilesDriver:         "local",
+		FilesPath:           "./files",
+		WebPath:             "/",
+	}
+
+	logger, err := mlog.NewLogger()
```

Comment:
You may find `logger := mlog.CreateConsoleTestLogger(true, mlog.LvlDebug)` to be more convenient for creating test loggers.